_examples/postcard/storage/simple_postgres_eventstore.go

```diff
@@ -18,6 +18,7 @@ func NewDefaultSimplePostcardRepository(ctx context.Context, db *sql.DB) (events
 		ctx,
 		db,
 		eventstore.NewPostgresSQLConfig[postcard.Postcard](
+			"PostcardSimpleDefault",
```

Comment:
Not really in scope for this PR, but I'm now wondering if it makes sense to move this to a method after all.

I'm thinking of something like:

```go
type TypedStream interface {
    StreamType() string
}
```

Which could be implemented or not (for simplicity). The repository would check if the stream implements the interface and either use the type or leave it blank.

Any thoughts? @maclav3 @roblaszczak

Reply:
for me, it could work nicely!

Reply:
👍

Reply:
Looks like it works pretty nicely https://github.com/ThreeDotsLabs/esja/pull/18/commits/d12d3b018e3fbb4d8a3c087417765d618615e80f

Reply:
Nice!

One suggestion to keep it more explicit: we could check the `AggregateType()` on the instance that's passed to the repository. It's available there, so we don't need to store it in the schema adapter as a field. If someone wants to, they could make the aggregate type dynamic then.

Reply:
I checked that and the problem is when we load events from the database. We have just an `ID` so we cannot resolve  `AggregateType`. Adding this as an additional parameter would be too much, don't you think?
https://github.com/ThreeDotsLabs/esja/blob/5a7d57e7074871591528268adb1b859379972640/eventstore/eventstore.go#L15-L22

Reply:
https://github.com/ThreeDotsLabs/esja/pull/18/commits/e6a9f48e4776bf6167019819f60a1b896a18208e